src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts

```diff
@@ -0,0 +1,74 @@
+import { Token } from '@uniswap/sdk-core'
+import { atom, useAtom } from 'jotai'
+import { loadable } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+const BASE_URL = 'https://temp.api.uniswap.org/v1/tokens/search?tokenQuery='
+
+type Address = `0x${string}`
+
+interface FetchTokensResult {
+  chainId: number
+  name: string
+  address: Address
+  decimals: number
+  symbol: string
+  logoURI: string
+  coinGeckoId: string
+  priceUsd: number
+  price24hChange: number
+  volume24h: number
+  marketCap: number
+}
+
+interface FetchTokensApiResult {
+  code: number
+  status: 'success' | 'failed'
+  hasNext: boolean
+  data: FetchTokensResult[]
+}
+
+async function fetchTokens(query: string): Promise<FetchTokensApiResult> {
+  const response = await fetch(`${BASE_URL}${query}`)
+  const data = await response.json()
+
+  return data
+}
+
+const searchQuery = atom('')
+
+const tokensData = loadable(
+  atom(async (get) => {
+    const query = get(searchQuery)
+
+    if (query.length > 0) {
+      return await fetchTokens(query)
+    }
+
+    return undefined
+  })
+)
+
+export function useTokenSearch(_query: string, existingTokens: Map<string, boolean>): Token[] {
+  const [query, setQuery] = useAtom(searchQuery)
+  // query is being used through jotai
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const [result] = useAtom(useMemo(() => tokensData, [query]))
```

Comment:
Nitpick: if only reading, you can use `useAtomValue` instead

Reply:
Great point - updated